### ISSUE TITLE
Pyfa 2 Certificate error reading ESI data

### ISSUE THREAD

**Issue by BoraVyvorant** [pyfa-org/Pyfa#1548]
## Bug Report

On initial startup in a clean environment (no `~/.pyfa` directory), Pyfa 2 starts but fails to read ESI information on use of "EVE SSO" --> "Manage Characters". No SSO functionality is available.

### Expected behavior:

Not an error message.

### Actual behavior:

![screen shot 2018-05-09 at 08 07 30](https://user-images.githubusercontent.com/12691802/39800838-1c003d12-5361-11e8-940e-f06fc0841d5a.png)

In the log file:

```
[2018-05-09 07:06:48.610329] ERROR: service.esi: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:777)>
```

### Detailed steps to reproduce:

Remove `~/.pyfa`. Start Pyfa. "EVE SSO" --> "Manage Characters".

### Fits involved in EFT format (Edit > To Clipboard > EFT):

n/a

### Release or development git branch? Please note the release version or commit hash:

0.0.0 (v2.0.0)

### Operating system and version (eg: Windows 10, OS X 10.9, OS X 10.11, Ubuntu 16.10):

macOS 10.13.4 (latest)

### Other relevant information:

If I try to proceed beyond this point using, say, "Manage Characters" again, I get a dialog but "Add Character" gives me an exception dialog as follows:

```
OS version: Darwin-17.5.0-x86_64-i386-64bit
Python version: 3.6.3 (default, Oct 22 2017, 22:24:50) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.72)]
wxPython version: 4.0.0b2 (wxWidgets 3.0.4)
SQLAlchemy version: 1.1.10
Logbook version: 1.0.0
Requests version: 2.18.4
Dateutil version: 2.6.1

####################

Traceback (most recent call last):
  File "/Users/ryan/pyfa/gui/esiFittings.py", line 329, in addChar
  File "/Users/ryan/pyfa/service/esi.py", line 209, in login
AttributeError: 'Esi' object has no attribute 'settings'
```

So this is really a blocker for me.

**Comment by blitzmann** [comment]
Hey there!

Hrm... interesting. It wasn't causing any issues in my test environment. Will take a look tonight. :)

**Comment by blitzmann** [comment]
Have you tried multiple times after restarting? Just wondering if it was a network fluke of some kind.

**Comment by BoraVyvorant** [comment]
Anything can _always_ be a network fluke, but I don't think this one is.

Issue is still present on the original machine. Persists through multiple invocations. Persists after reboot. Happens just the same on a second machine.

Is there anything I can do to get more diagnostics out of this?  Seems like it's an issue with whatever bundle of PKI roots it's using, but I don't know whether the application ships with one or if it's using something from the environment.  Obviously if we had some debug saying which endpoint it was talking to, and which certificate list it used, that might help.

**Comment by blitzmann** [comment]
That message pops up here: 

https://github.com/pyfa-org/Pyfa/blob/master/service/esi.py#L92

So, it's happening when EsiPy (using pyswagger) attempts to fetch information for the ESI endpoints. From there, I'm not really sure how to debug that.

I should maybe mention: I supply the certs that `requests` needs:
https://github.com/pyfa-org/Pyfa/blob/master/dist_assets/mac/pyfa.spec#L23

I'm not sure if this is really required anymore or what. used to be required for 1.x due to exactly this issue...

**Comment by Fundaris** [comment]
Could it be the move of https://esi.tech.ccp.is to https://esi.evetech.net ?
I'm not good at python nor have I a mac x)

**Comment by BoraVyvorant** [comment]
@blitzmann I think I read in another issue somewhere that you aren't using a current version of macOS for development; is that correct? If so, I might try firing up some old versions in VMs and see what happens. Might be able to look at this later in the week.

**Comment by blitzmann** [comment]
> Could it be the move of https://esi.tech.ccp.is to https://esi.evetech.net ?

This move was accounted for

> I think I read in another issue somewhere that you aren't using a current version of macOS for development; is that correct?

I'm using 10.10 I believe. The reason being for backwards compatibility (if I build on 10.12, it's not likely to be able to run on anything below that).

Also, I'm not a mac guy in the slightest. I have an old core 2 duo mac mini running 10.10 that I use as my builder lol. 

I am going to release a point release hopefully tonight that reworks the ESI stuff, so I'm hoping that takes care of this. it's a shot in the dark, though I hope my aim is lucky ;)

**Comment by blitzmann** [comment]
New release is out, can you check to see if you're still experiencing the same issue?

**Comment by BoraVyvorant** [comment]
This issue appears to be resolved for me in 2.0.1.

I hit a couple of unrelated problems when experimenting (this is the first time I've been able to do anything with v2) but I will create new issues for those.

Thanks!

**Comment by blitzmann** [comment]
Excellent! It think it may have been because we were using pyswagger, which may have been doing something funky with the certs it requires. Glad to hear it's working now :)

>I hit a couple of unrelated problems when experimenting (this is the first time I've been able to do anything with v2) but I will create new issues for those.

Please do! 

Gonna close this as resolved
